### PR TITLE
Product and Variant bulk mutations

### PR DESCRIPTION
**Pull request by SzymJ**
This base branch contains changes related to Product and Variant bulk mutations proposed in #10798 RFC.

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [x] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/product/bulk_mutations/category_bulk_delete.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        model = models.Category
+        object_type = Category
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductError
+        error_type_field = "product_errors"
+
+    @classmethod
+    def bulk_action(cls, info, queryset):
+        manager = load_plugin_manager(info.context)
+        delete_categories(queryset.values_list("pk", flat=True), manager)
```
Comment (maarcingebala) [inline saleor/graphql/product/bulk_mutations/category_bulk_delete.py:29]: This function triggers `CATEGORY_DELETED` and `PRODUCT_UPDATED` webhooks for all removed categories and each of their products. Is that what we want @patrys? Setting product's category to null also results in making this product unpublished within a channel.

Comment (patrys) [reply saleor/graphql/product/bulk_mutations/category_bulk_delete.py:29]: I don't think this should fire both. We must decide if the product/category association belongs to the product or the category.

As for unpublishing, we want to make categories optional (and possibly allow multiple categories to be assigned), so I don't think it should result in the product becoming unpublished. How does the API behave if a client requests the category field and no category is assigned? Do we return `null`? If so, we can make the category optional when this is merged.

File: saleor/graphql/product/bulk_mutations/product_type_bulk_delete.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+from ....attribute import models as attribute_models
+from ....core.permissions import ProductTypePermissions
+from ....core.tracing import traced_atomic_transaction
+from ....product import models
+from ...core.mutations import ModelBulkDeleteMutation
+from ...core.types import NonNullList, ProductError
+from ..types import ProductType
+
+
+class ProductTypeBulkDelete(ModelBulkDeleteMutation):
```
[the remark this answers is not in the document]
Comment (SzymJ) [reply saleor/graphql/product/bulk_mutations/product_type_bulk_delete.py:16]: Not related to product bulk mutations, I will create a separate issue for that.

File: saleor/graphql/product/bulk_mutations/collection_bulk_delete.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            .filter(collections__in=collections_ids)
+            .distinct()
+        )
+        manager = load_plugin_manager(info.context)
+        for collection in queryset.iterator():
+            manager.collection_deleted(collection)
+        queryset.delete()
+
+        for product in products:
+            manager.product_updated(product)
```
[the remark this answers is not in the document]
Comment (SzymJ) [reply saleor/graphql/product/bulk_mutations/collection_bulk_delete.py:39]: Not related to product bulk mutations, I will create a separate issue for that.

File: saleor/graphql/product/bulk_mutations/category_bulk_delete.py (hunk 1 above)
Comment (SzymJ) [reply saleor/graphql/product/bulk_mutations/category_bulk_delete.py:29]: Not related to product bulk mutations, I will create a separate issue for that.

**Review by maarcingebala — APPROVED** [review]

**Review by kadewu — APPROVED** [review]

**Pull request merged**